src/lib/core/__test__/candidate.unit.test.ts

```diff
@@ -1,50 +1,104 @@
-      submissionData,
-    )
+    const param: any = {
+      userId: uuid,
+      quizId: uuid,
+      code: text,
+    }
+    prismaMock.submission.create.mockResolvedValue(param)
+    expect(await createCandidateQuizSubmission(param)).toEqual(param)
   })
 
   test('Missing userId parameter should raise an missing userId error', async () => {
-    const submissionData: any = {
-      // userId: faker.string.uuid(),
-      quizId: faker.string.uuid(),
-      code: faker.lorem.text(),
+    const param: any = {
+      // userId: uuid,
+      quizId: uuid,
+      code: text,
     }
     expect(
-      async () => await createCandidateQuizSubmission(submissionData),
+      async () => await createCandidateQuizSubmission(param),
     ).rejects.toThrow(/^missing userId$/)
   })
 
   test('Missing quizId parameter should raise an missing quizId error', async () => {
-    const submissionData: any = {
-      userId: faker.string.uuid(),
-      // quizId: faker.string.uuid(),
-      code: faker.lorem.text(),
+    const param: any = {
+      userId: uuid,
+      // quizId: uuid,
+      code: text,
     }
     expect(
-      async () => await createCandidateQuizSubmission(submissionData),
+      async () => await createCandidateQuizSubmission(param),
     ).rejects.toThrow(/^missing quizId$/)
   })
 
   test('Missing code parameter should raise an missing code error', async () => {
-    const submissionData: any = {
-      userId: faker.string.uuid(),
-      quizId: faker.string.uuid(),
-      // code: faker.lorem.text(),
+    const param: any = {
+      userId: uuid,
+      quizId: uuid,
+      // code: text,
     }
     expect(
-      async () => await createCandidateQuizSubmission(submissionData),
+      async () => await createCandidateQuizSubmission(param),
     ).rejects.toThrow(/^missing code$/)
   })
+
+  test('getActivityLogCount fn should return the count number', async () => {
+    const param: any = {
+      assessmentIds: [uuid, uuid],
+    }
+    const prismaMockValue = 5
+    prismaMock.candidateActivityLog.count.mockResolvedValue(prismaMockValue)
+    expect(await getActivityLogCount(param)).toBe(prismaMockValue)
+  })
+
+  test('Missing assessmentIds parameter should raise an missing assessmentIds error', async () => {
+    const param: any = {}
+    expect(async () => await getActivityLogCount(param)).rejects.toThrow(
+      /^missing assessmentIds$/,
+    )
+  })
+
+  test('Empty assessmentIds should raise an empty assessmentIds error', async () => {
+    const param: any = {
+      assessmentIds: [],
+    }
+    expect(async () => await getActivityLogCount(param)).rejects.toThrow(
+      /^empty assessmentIds$/,
+    )
+  })
+
+  test('getActivityLogs fn should return activity logs', async () => {
+    const param: any = {
+      assessmentIds: [uuid, uuid],
+    }
+    prismaMock.candidateActivityLog.findMany.mockResolvedValue(param)
+    expect(await getActivityLogs(param)).toEqual(param)
```

Comment:
This test may not be comprehensive. Can't tell what are the exact return data we expect, we should test the existence of all the return fields. Mock data should be as close as the real data. With this test, if we change the logic and return less two or three fields, this test will give us a false pass.